src/plugins/memdump/memdump.cpp

```diff
@@ -233,42 +268,46 @@ bool dump_memory_region(
 
     chk_str = g_checksum_get_string(checksum);
 
-    if (asprintf(&file, "%llx_%.16s", (unsigned long long) ctx->addr, chk_str) < 0)
+    if (asprintf(&file, "memdump.%06d", ++plugin->dumps_count) < 0)
```

Comment:
The original file name format is `<dump base address>_<content hash>`, which yields the following (pretty implicit) features:
* When disassembling a dump file, you don't need to consult any external sources, you can deduce the dump base address (required for disassembling) from the file name itself, which is very useful for both manual and automatic processing
* Sometimes is happens that the same buffer is dumped multiple times - each consecutive memory dump would get the same file name. I remember that this is extremely important or otherwise you may get lots of files with the same content (e.g. each file generated by different heuristic).
* (We rely on this format in all our tools, thus compatibility would be broken. This is not a very big issue but I would prefer not to break compatibility unless a very good reason appears.)

In general I support the idea of saving metadata together with the dumps, but it would be great to have it in a backwards-compatible way.

Reply:
> When disassembling a dump file, you don't need to consult any external sources

Usually I need PID and operation too.

> We rely on this format in all our tools

I rely on the format in my tools too :)

> it would be great to have it in a backwards-compatible way.

I will add command line argument to select the format. Your format would be the default one.

Reply:
@icedevml I have add command line arugemnt `--compatible-file-name` to resolve the issue.

Reply:
This is introducing two formats at the same time, maybe we could agree on a single one that would satisfy everyone? I will get back to it in a few days.